assets/blocks/course-outline/store.js

```diff
@@ -39,13 +40,14 @@ const actions = {
 			} );
 			yield actions.setStructure( result );
 		} catch ( error ) {
-			yield dispatch( 'core/notices' ).createErrorNotice(
-				[
-					__( 'Course Outline update failed:', 'sensei-lms' ),
-					error.message,
-				].join( ' ' )
+			const errorMessage = sprintf(
+				/* translators: Error message. */
+				__( 'The course could not be saved. %s', 'sensei-lms' ),
+				error.message
 			);
-			yield actions.setEditorDirty( false );
```

Comment:
I removed the dirty clear from the error catch because if it didn't save, it continues dirty.
It also allows the user to resave in case of a server instability error.